### PR TITLE
feat(vod): add new resource watermark template

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

add new resource watermark template

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add new resource watermark template
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/vod' TESTARGS='-run TestAccWatermarkTemplate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vod -v -run TestAccWatermarkTemplate_basic -timeout 360m -parallel 4
=== RUN   TestAccWatermarkTemplate_basic
=== PAUSE TestAccWatermarkTemplate_basic
=== CONT  TestAccWatermarkTemplate_basic
--- PASS: TestAccWatermarkTemplate_basic (25.05s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vod       25.143s
```

### CODE REVIEW

File: docs/resources/vod_watermark_template.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+* `height` - (Optional, String) Specifies height ratio of the watermark image relative to the output video.
+  The value range is (0, 1). It supports 4 decimal places, e.g. 0.9999, the excess will be
+  automatically discarded. Defaults to: **0.01**.
+
+* `timeline_start` - (Optional, String) Specifies the watermark start time (Unit: second). Defaults to: **0**.
+
+* `timeline_duration` - (Optional, String) Specifies the watermark duration (Unit: second). The range is
+  [`timeline_start`, ToEND]. 'ToEND' means continue until the end of the video. Defaults to: **ToEND**.
+
+Default: ToEND.
```
Comment (niuzhenguo) [inline docs/resources/vod_watermark_template.md]: duplicate with description above.

File: huaweicloud/provider.go [hunk 2]
```diff
@@ -850,7 +850,8 @@ func Provider() *schema.Provider {
 			"huaweicloud_iam_agency":    iam.ResourceIAMAgencyV3(),
 			"huaweicloud_iam_agency_v3": iam.ResourceIAMAgencyV3(),
 
-			"huaweicloud_vod_media_category": vod.ResourceMediaCategory(),
+			"huaweicloud_vod_media_category":     vod.ResourceMediaCategory(),
```
Comment (niuzhenguo) [inline huaweicloud/provider.go]: should move this above, this is Legacy part

File: huaweicloud/provider.go [hunk 3]
```diff
@@ -850,7 +850,8 @@ func Provider() *schema.Provider {
...
+			"huaweicloud_vod_watermark_template": vod.ResourceWatermarkTemplate(),
```
Comment (niuzhenguo) [inline huaweicloud/provider.go]: ditto

File: huaweicloud/services/vod/resource_huaweicloud_vod_watermark_template.go [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+	}
+
+	resp, err := client.CreateWatermarkTemplate(&createReq)
+	if err != nil {
+		return diag.Errorf("error creating VOD watermark template: %s", err)
+	}
+
+	err = uploadImage(*resp.UploadUrl, d.Get("image_file").(string))
+	if err != nil {
+		return diag.Errorf("error uploading watermark image: %s", err)
```
Comment (niuzhenguo) [inline huaweicloud/services/vod/resource_huaweicloud_vod_watermark_template.go:206]: should we move d.SetId above this uploadImage operation? if uploadImage or confirmImageUpload request errored, the resource created will be a dangling resource.

File: docs/resources/vod_watermark_template.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/vod_watermark_template.md]: fixed

File: huaweicloud/provider.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/provider.go]: fixed

File: huaweicloud/provider.go (hunk 3 above)
Comment (Jason-Zhang9309) [reply huaweicloud/provider.go]: fixed

File: huaweicloud/services/vod/resource_huaweicloud_vod_watermark_template.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/vod/resource_huaweicloud_vod_watermark_template.go:206]: sure

File: docs/resources/vod_watermark_template.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  automatically discarded. Defaults to: **0.01**.
+
+* `height` - (Optional, String) Specifies height ratio of the watermark image relative to the output video.
+  The value range is (0, 1). It supports 4 decimal places, e.g. 0.9999, the excess will be
+  automatically discarded. Defaults to: **0.01**.
+
+* `timeline_start` - (Optional, String) Specifies the watermark start time (Unit: second). Defaults to: **0**.
+
+* `timeline_duration` - (Optional, String) Specifies the watermark duration (Unit: second). The range is
+  [`timeline_start`, ToEND]. 'ToEND' means continue until the end of the video. Defaults to: **ToEND**.
```
Comment (hmbig2) [inline docs/resources/vod_watermark_template.md]: s/ToEND/`ToEND`

File: docs/resources/vod_watermark_template.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource. If omitted, the
+  provider-level region will be used. Changing this creates a new resource.
+
+* `name` - (Required, String) Specifies the template name, which contains a maximum of 128 characters.
+
+* `image_file` - (Required, String, ForceNew) Specifies the image file name, e.g. './test.png'.
+
+* `image_type` - (Required, String, ForceNew) Specifies the image file type. The valid values are **PNG**, **JPG**
+  and **JPEG**.
```
Comment (ShiChangkuo) [inline docs/resources/vod_watermark_template.md]: please add `Changing this creates a new resource.` for ForceNew parameters

File: docs/resources/vod_watermark_template.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+* `image_process` - (Optional, String, ForceNew) Specifies the image process. The valid values are:
```
Comment (ShiChangkuo) [inline docs/resources/vod_watermark_template.md]: this param can be updated, please check it

File: huaweicloud/services/vod/resource_huaweicloud_vod_watermark_template.go [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+		return err
+	}
+	defer data.Close()
+	req, err := http.NewRequest("PUT", uploadUrl, data)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "text/plain")
+
+	httpClient := &http.Client{}
```
Comment (ShiChangkuo) [inline huaweicloud/services/vod/resource_huaweicloud_vod_watermark_template.go]: please add Timeout for http client

Comment (Jason-Zhang9309) [reply huaweicloud/services/vod/resource_huaweicloud_vod_watermark_template.go]: fixed

File: docs/resources/vod_watermark_template.md (hunk 7 above)
Comment (Jason-Zhang9309) [reply docs/resources/vod_watermark_template.md]: fixed

File: docs/resources/vod_watermark_template.md (hunk 5 above)
Comment (Jason-Zhang9309) [reply docs/resources/vod_watermark_template.md]: use another way to describe

File: docs/resources/vod_watermark_template.md (hunk 6 above)
Comment (Jason-Zhang9309) [reply docs/resources/vod_watermark_template.md]: fixed

**Pull request merged**